pom.xml

```diff
@@ -669,12 +692,36 @@
 
     <profiles>
         <profile>
+            <id>non-jigsaw</id>
+            <activation>
+                <!-- The versions we're using of Ehcache, Infinispan and Neo4J won't work on JDK9. Needs upgrading.  -->
+                <jdk>[1.7,1.8]</jdk>
+            </activation>
+            <modules>
+                <module>ehcache</module>
```

Comment:
Regarding Infinispan and Neo4J they have global issues.

For ehcache we could exclude a single test but it doesn't give us much value: either way, we can't say it will work on Java9 so I think the better approach is to defer this to an Ehcache upgrade, or a full investigation of why/what Ehcache is going to do with Java9.

In general I'd prefer to merge this as "the core builds", and open follow-up investigations so we can break it up in smaller tasks.